File: Mcs-cs/Mcs/Models/Configuration.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace Mcs.Models
{
    public class Configuration
    {
        public readonly AlgorithmType AlgorithmType;
        public readonly GeneticConfiguration GeneticConfiguration;
        public readonly string InputFilePath;

        public Configuration(AlgorithmType algorithmType, GeneticConfiguration geneticConfiguration, string inputFilePath)
        {
            AlgorithmType = algorithmType;
            GeneticConfiguration = geneticConfiguration;
            InputFilePath = inputFilePath;
        }

        public Configuration(AlgorithmType algorithmType, string inputFilePath) : this(algorithmType, null, inputFilePath)
        {
        }
    }
}

File: mcs-java/src/main/java/anc/models/Configuration.java
package anc.models;

public class Configuration {
    public final ProblemType problemType;
    public final AlgorithmType algorithmType;
    public final GeneticConfiguration geneticConfiguration;
    public final String inputFilePath;

    public Configuration(ProblemType problemType, AlgorithmType algorithmType, String inputFilePath) {
        this.problemType = problemType;
        this.algorithmType = algorithmType;
        this.geneticConfiguration = null;
        this.inputFilePath = inputFilePath;
    }

    public Configuration(ProblemType problemType, AlgorithmType algorithmType, GeneticConfiguration geneticConfiguration, String inputFilePath) {
        this.problemType = problemType;
        this.algorithmType = algorithmType;
        this.geneticConfiguration = geneticConfiguration;
        this.inputFilePath = inputFilePath;
    }
}
